content/blog/2019-09-25-RecordTypes.md

```diff
@@ -95,29 +95,73 @@ This syntax was designed to have a similar format as record instantiation.
+    Person Henry = { 20, false };
+    updateAge(Henry);
+    return 0;
+}
+```
+Compiling this program to Bril may look something like this, assuming we ignore function calls.
+```
+type Person = {age: int; isAsleep: bool};
+v0: int = const 20;
+v1: bool = const false;
+Henry: Person = record {age: v0; isAsleep: v1};
+v2: int = const 21;
+Henry: Person = Henry with {age: v2};
+```
+As shown in this example, mutable record types can be compiled into immutable records in Bril without significant effort. Therefore, lack of this operation does not compromise this goal.
+
+Finally, to evaluate record types as a language feature in Bril, we consider how this functionality translates to Bril.
 We found that creating new records was a tedious process if the record was large, so we implemented *with* statements in addition to the features mentioned above for situations where one wanted to duplicate a record with a few changes. It should be noted that it is bad form to use a with statement with no fields because that would be identical to referencing the old record with `… = id oldRecordName`.
 
-We were successful in this aspect as creating a new record from an existing one but changing one field is nearly the same amount of code as mutating directly.
+We were successful in this aspect as creating a new record from an existing one with this syntax is more concise and easier to reason about than copying over every field. This is an advantage in an IR as we can logically think about a single _with statement_ and a sequence of functionally equivalent statements that copy variables in the same way.
+
+Consider the blocks of code below. These programs duplicate a record with one field changed.
+Here we show what this would look like without _with statements_. 
+
 ```
-type Person = {age: int; isAsleep: bool}
+type Person = {age: int; isHappy: bool};
 v0: int = const 21;
 v1: bool = const false;
-Henry: Person = record {age: v0; isAsleep: v1}
+Henry: Person = record {age: v0; isHappy: v1};
 v2: bool = const true;
-AwakeHenry: Person = Henry with {isAsleep: v2}
+v3: int = Henry . age;
+AwakeHenry: Person = record {age: v3; isAsleep: v2};
 ```
-Possible Mutable Record Syntax:
+Here we use the with syntax.
 ```
-type Person = {age: int; isHappy: bool}
+type Person = {age: int; isAsleep: bool};
 v0: int = const 21;
 v1: bool = const false;
-Henry: Person = record {age: v0; isHappy: v1}
+Henry: Person = record {age: v0; isAsleep: v1};
 v2: bool = const true;
-Henry.isHappy = v2
+AwakeHenry: Person = Henry with {isAsleep: v2};
 ```
+It is worth noting that the size of code required to duplicate a record without _with statements_ scales linearly with the size of the record. In contrast, the size of code required to duplicate a record with _with statements_ does not change as the size of the record increases. Therefore, record types are successsful as a language feature as they integrate well with current syntax and do not impose unneccesary code bloat. 
```

Comment:
```suggestion
It is worth noting that the size of code required to duplicate a record without _with statements_ scales linearly with the size of the record. In contrast, the size of code required to duplicate a record with _with statements_ only increases with the size of the changes. Therefore, record types are successful as a language feature as they integrate well with current syntax and do not impose unnecessary code bloat. 
```
I think it's important to note that the size of the with instruction is not technically unchanged as one would have to write more field-value-pairs if they want more changes. This would also be reflected into whatever assembly Bril is compiled into.

Reply:
hmm both both with statements and without with statements scale in that way as well. Maybe we can mention they are the same in that regard, but the difference is that using with statements does not change in size as you increase number of record fields